### PR TITLE
SUS-1719: fix caching issues with Wall

### PR DESCRIPTION
**Pull request by mszabo-wikia**
https://wikia-inc.atlassian.net/browse/SUS-1719

### CODE REVIEW

File: extensions/wikia/Wall/WallHooksHelper.class.php [hunk 1]
```diff
@@ -667,30 +667,6 @@ static public function onArticleDelete( $article, &$user, &$reason, &$error ) {
 	}
 
 	/**
-	 * @param RecentChange $recentChange
-	 * @return bool
-	 */
-	static public function onRecentChangeSave( $recentChange ) {
```
Comment (rybmat) [inline extensions/wikia/Wall/WallHooksHelper.class.php:673]: @TK-999 what is the reason for removing this hook? At the first sight it doesn't look related to bug reported in the ticket

File: extensions/wikia/Wall/builders/WallEditBuilder.class.php [hunk 2]
```diff
@@ -30,17 +30,23 @@ public function editWallMessage(): WallEditBuilder {
 		 */
 		Hooks::register( 'ArticleDoEdit', [ $this, 'updateCommentsIndexEntry' ] );
 
-		$result = $this->message->getArticleComment()->doSaveComment( $this->messageText, $this->editor );
-		if ( !$result ) {
+		$comment = $this->message->getArticleComment();
+		$result = $comment->doSaveComment( $this->messageText, $this->editor );
+		if ( !$result || !$result[0]->isOK() ) {
```
Comment (rybmat) [inline extensions/wikia/Wall/builders/WallEditBuilder.class.php:36]: 👍  I fixed it also in SUS-1189 😄

File: extensions/wikia/Wall/WallHooksHelper.class.php (hunk 1 above)
Comment (mszabo-wikia) [reply extensions/wikia/Wall/WallHooksHelper.class.php:673]: This refactors how wall notifications are generated when message is edited - now we directly generate notification from `EditBuilder`, so that we can pass the new `Revision` instance instead of having to query the database.

For Wall custom actions, `rc_this_oldid` is always 0, so this never worked.

Comment (rybmat) [reply extensions/wikia/Wall/WallHooksHelper.class.php:673]: 👍

**Comment by rybmat** [comment]
https://github.com/Wikia/app/blob/7cb74698306c977063b7d3469b42f0f13f9bba4e/extensions/wikia/Wall/WallMessage.class.php#L1109-L1109

looks like this line is not used and could be removed

File: extensions/wikia/Wall/WallMessage.class.php [hunk 3]
```diff
@@ -683,32 +683,6 @@ public function getMessagePageUrl( $withoutAnchor = false, $fullUrl = true ) {
 		return $this->messagePageUrl[ $withoutAnchor ];
 	}
 
-	public function getArticleId( &$articleData = null ) {
-		$title = $this->getArticleComment()->getTitle();
-		$articleId = $this->getArticleComment()->getTitle()->getArticleId();
-
-		if ( $articleId === false ) {
-			Wikia::log( __METHOD__, false, "WALL_NO_ARTILE_ID" . print_r( [ '$title' => $title ], true ) );
```
Comment (rybmat) [inline extensions/wikia/Wall/WallMessage.class.php:691]: don't we want to log such events? getId() does not do this

**Review by rybmat — APPROVED** [review]

File: extensions/wikia/Wall/WallMessage.class.php (hunk 3 above)
Comment (mszabo-wikia) [reply extensions/wikia/Wall/WallMessage.class.php:691]: In Kibana there are no log entries of this, and this method was only used in 2 places: in builder and in RC hook handler.

File: extensions/wikia/Wall/builders/WallMessageBuilder.class.php [hunk 4]
```diff
@@ -198,20 +198,24 @@ public function notifyEveryoneForNewThreadIfNeeded(): WallMessageBuilder {
...
+				( new CommentsIndexEntry() )
+					->setNamespace( $title->getNamespace() )
+					->setParentPageId( $parentPageId )
+					->setParentCommentId( $parentCommentId )
+					->setCommentId( $title->getArticleID() )
+					->setFirstRevId( $revId )
+					->setLastRevId( $revId );
+
+			return CommentsIndex::getInstance()->insertEntry( $entry, $dbw );
+		}
```
Comment (macbre) [inline extensions/wikia/Wall/builders/WallMessageBuilder.class.php]: Let's log warnings when this check fails to identify incorrect calls.

File: extensions/wikia/Wall/builders/WallEditBuilder.class.php [hunk 5]
```diff
@@ -64,10 +64,12 @@ public function editWallMessage(): WallEditBuilder {
...
 	public function updateCommentsIndexEntry( DatabaseBase $dbw, Title $title, Revision $rev ): bool {
-		$entry = CommentsIndex::getInstance()->entryFromId( $title->getArticleID() );
-		$entry->setLastRevId( $rev->getId() );
+		if ( $title->isTalkPage() && WallHelper::isWallNamespace( $title->getNamespace() ) ) {
+			$entry = CommentsIndex::getInstance()->entryFromId( $title->getArticleID() );
+			$entry->setLastRevId( $rev->getId() );
 
-		CommentsIndex::getInstance()->updateEntry( $entry, $dbw );
+			CommentsIndex::getInstance()->updateEntry( $entry, $dbw );
+		}
```
Comment (macbre) [inline extensions/wikia/Wall/builders/WallEditBuilder.class.php]: Let's log warnings when this check fails to identify incorrect calls.

File: extensions/wikia/Wall/builders/WallMessageBuilder.class.php [hunk 6]
```diff
@@ -97,10 +100,19 @@ public function postNewMessage(): WallMessageBuilder {
...
+		 * @var Article $article
+		 * @var Status $status
+		 * @var Revision $rev
+		 */
 		list( $status, $article ) = $result;
-		$this->newMessage = WallMessage::newFromTitle( $article->getTitle() );
+		$title = $article->getTitle();
+		$rev = $status->revision;
 
+		// Preload article ID - saves DB call
```
Comment (macbre) [inline extensions/wikia/Wall/builders/WallMessageBuilder.class.php:112]: :+1:

File: extensions/wikia/Wall/index/CommentsIndex.class.php [hunk 7]
```diff
@@ -62,11 +66,12 @@ public function updateEntry( CommentsIndexEntry $entry, DatabaseBase $dbw = null
 
 			$parentEntry = static::entryFromId( $parentId );
 			$parentEntry->setLastChildCommentId( $lastCommentId );
-			$this->updateEntry( $parentEntry );
+			$updateSuccess &= $this->updateEntry( $parentEntry );
```
Comment (macbre) [inline extensions/wikia/Wall/index/CommentsIndex.class.php:69]: `=` operator is enough here.

**Review by macbre — APPROVED** [review]
